File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_MessageArg.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_MessageArg
*/
public class t_MessageArg : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public sbyte	m_eType;
	public int	m_value;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_MessageArg() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_eType = 0;
		m_value = 0;
	}
	public bool read(CReadVariable cVariable) {
		try {
			m_eType = cVariable.getS8();
			m_value = cVariable.getS32();
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(m_eType);
		cVariable.put(m_value);
		return true;
	}
};

File: Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_MessageArg.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
#/==========================================================================
#*!
#	@brief	t_MessageArg
#/
class t_MessageArg:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_eType = 0	#S8
	m_value = 0	#S32
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_eType = 0
		self.m_value = 0

	def read(self,cVariable):
		self.m_eType = cVariable.getS8()
		self.m_value = cVariable.getS32()
		return True

	def write(self,cVariable):
		cVariable.putS8(self.m_eType)
		cVariable.putS32(self.m_value)
		return True

